velox/functions/sparksql/DecimalArithmetic.cpp

```diff
@@ -33,6 +33,175 @@ std::string getResultScale(std::string precision, std::string scale) {
       scale);
 }
 
+// Returns the whole and fraction parts of a decimal value.
+template <typename T>
+inline std::pair<T, T> getWholeAndFraction(T value, uint8_t scale) {
+  const auto scaleFactor = velox::DecimalUtil::kPowersOfTen[scale];
+  const T whole = value / scaleFactor;
+  return {whole, value - whole * scaleFactor};
+}
+
+// Increases the scale of input value by 'delta'. Returns the input value if
+// delta is not positive.
+inline int128_t increaseScale(int128_t in, int16_t delta) {
+  // No need to consider overflow as 'delta == higher scale - input scale', so
+  // the scaled value will not exceed the maximum of long decimal.
+  return delta <= 0 ? in : in * velox::DecimalUtil::kPowersOfTen[delta];
+}
+
+// Scales up the whole part to result scale, and combine it with fraction part
+// to produce a full result for decimal add. Checks whether the result
+// overflows.
+template <typename T>
+inline T
+decimalAddResult(T whole, T fraction, uint8_t resultScale, bool& overflow) {
+  T scaledWhole = DecimalUtil::multiply<T>(
+      whole, velox::DecimalUtil::kPowersOfTen[resultScale], overflow);
+  if UNLIKELY (overflow) {
```

Comment:
`UNLIKELY` need to be in brackets: `if (FOLLY_UNLIKELY(overflow))`

Reply:
Thanks for your comment. Fixed.